agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -121,9 +121,9 @@ bool slave_thread::init()
 
     if (config.ucc_listener_slave_hostap_iface == config.hostap_iface &&
         config.ucc_listener_port != 0) {
-        m_agent_ucc_listener =
-            std::make_unique<agent_ucc_listener>(config.ucc_listener_port, config.vendor,
-                                                 config.model, config.bridge_iface, &master_socket);
+        m_agent_ucc_listener = std::make_unique<agent_ucc_listener>(
+            config.ucc_listener_port, config.vendor, config.model, config.bridge_iface,
+            (Socket **)&master_socket);
```

Comment:
A better solution is to change the type of the controller socket to `SocketClient` in the `agent_ucc_listener` class.
Already did it in the UGW Support branch, so can be fixed later.

Reply:
ok, so once you merge the UGW Support branch you can override it.